redhat-developer/odo
# propagate deletes during `odo watch`

Watch fixes + propogate deletes of watched sources
  
This PR does the following:
1. fixes the use case of odo watch pushing observed changes
   from any level nested in source root directory to
   the component source root because of which, the changes
   are not reflected in the component.
2. Enables propogation of delete of watched source files/folders
3. Adds more e2e tests for watch
4. Covers newly added modules via UTs
5. Enhances openjdk sample credits 
Steps to verify this PR:
Test 1:
  a. trigger odo watch on a local source
  b. make change in a file in a nested dir in the source directory
  c. See the change is reflected in the component(perhaps via component url? :) )
Test 2:
Steps in #950
    
fixes #1099, #1054, #988, #1121, #1117
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>
